packages/p2p/src/stores/general-store.js

```diff
@@ -842,6 +842,7 @@ export default class GeneralStore extends BaseStore {
         p2p_settings[this.client.loginid] = user_settings;
 
         localStorage.setItem('p2p_settings', JSON.stringify(p2p_settings));
+        window.dispatchEvent(new Event('storage'));
```

Comment:
Has to dispatch an event with the `storage` type so it notifies the listeners of storage changes.